src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -8874,4 +8835,59 @@ void GatherComponentLoadsSurfAbsFact(EnergyPlusData &state)
     }
 }
 
+void InitSurfacePropertyViewFactors(EnergyPlusData &state)
+{
+
+    // purpuse:
+    //   Initializes sky and ground surfaces view factors of exterior surfaces
+    //   used by SurfaceProperty:LocalEnvironment
+    //   view factors are constant hence should be set only once
+
+    if (!state.dataGlobal->AnyLocalEnvironmentsInModel) {
+        return;
+    }
+    if (!state.dataHeatBalSurfMgr->InitSurfaceHeatBalancefirstTime) {
+        return;
+    }
+    for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
+        if (!state.dataSurface->SurfHasSurroundingSurfProperties(SurfNum)) continue;
+        auto &Surface = state.dataSurface->Surface(SurfNum);
+        int SrdSurfsNum = state.dataSurface->SurfSurroundingSurfacesNum(SurfNum);
+        auto &SrdSurfsProperty = state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum);
+        Real64 SrdSurfsViewFactor = 0;
+
+        SrdSurfsViewFactor += SrdSurfsProperty.SkyViewFactor;
+        SrdSurfsViewFactor += SrdSurfsProperty.GroundViewFactor;
+        for (int SrdSurfNum = 1; SrdSurfNum <= SrdSurfsProperty.TotSurroundingSurface; SrdSurfNum++) {
+            SrdSurfsViewFactor += SrdSurfsProperty.SurroundingSurfs(SrdSurfNum).ViewFactor;
+        }
+        // Check if the sum of all defined view factors > 1.0
+        if (SrdSurfsViewFactor > 1.0) {
+            ShowSevereError(state, "Illegal surrounding surfaces view factors for " + Surface.Name + ".");
+            ShowContinueError(state, " The sum of sky, ground, and all surrounding surfaces view factors should be less than or equal to 1.0.");
+        }
+        if (SrdSurfsProperty.SkyViewFactor > 0.0 && SrdSurfsProperty.GroundViewFactor > 0.0) {
+            // If both surface sky and ground view factor defined, overwrite with the defined value
+            Surface.ViewFactorSkyIR = SrdSurfsProperty.SkyViewFactor;
+            Surface.ViewFactorGroundIR = SrdSurfsProperty.GroundViewFactor;
+        } else if (SrdSurfsProperty.SkyViewFactor > 0.0 && SrdSurfsProperty.GroundViewFactor == 0.0) {
```

Comment:
@Nigusse Zero is not the same as blank in the current code:
`GroundViewFactor` and `SkyViewFactor` were initialized to -1, then in [`GetSurfaceSrdSurfsData`](https://github.com/NREL/EnergyPlus/blob/43b09606e2a93a0a1b220949d5b6832d63c0a03a/src/EnergyPlus/SurfaceGeometry.cc#L8179-L8192) if the input field is blank, the value remains at -1 and that property is left at the normal value. But if the user input a zero value, then the value would be set to zero.

This isn't a great design, especially since the IDD defaults are 0.5. So I'm not sure what the best solution is here, but the changes here alter the behavior.

Reply:
How does the revised code distinguish between blank and an input value of 0.0?